src/nix-build/nix-build.cc

```diff
@@ -417,16 +417,19 @@ void mainWrapped(int argc, char * * argv)
                 "dontAddDisableDepTrack=1; "
                 "[ -e $stdenv/setup ] && source $stdenv/setup; "
                 "%3%"
+                "PATH=\"$(dirname %4%):$PATH\"; "
```

Comment:
This can be `PATH=%N%` where N = `dirOf(shell)`. I.e. compute the directory in C++, saving an external call.

Reply:
Ah, nice! Fixed!